ASIT-003

Bagietek opened the pull request:
closes #3

Bagietek committed:
ASIT-003

Get document by UUID endpoint

Bagietek committed:
Zmiana z wyświetlania na pobieranie

Zmieniono endpoint tak aby pobierał plik zamiast wyświetlał go w oknie przeglądarki

anastasimars commented on src/main/java/pl/akademiaspecjalistowit/DocumentFlowManagementService/document/service/DocumentService.java:
tutaj lepiej byłoby nazwać to downloadDocument()

anastasimars commented on src/main/java/pl/akademiaspecjalistowit/DocumentFlowManagementService/document/DocumentController.java:
```diff
@@ -28,4 +26,14 @@ ResponseEntity<UUID> uploadDocument(@RequestParam("file") MultipartFile file){
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_PDF);
+        //todo: po zaktualizowaniu pól DocumentEntity przypisać poprawną nazwę pobieranego pliku
+        headers.setContentDisposition(ContentDisposition.builder("attachment").filename("document.pdf").build());
```
Przemek chyba mówił, że ta metoda musi automatycznie pobierać dokument. A u mnie tylko pokazywała dokument. Kiedy zmieniłam headers.setContentDisposition(ContentDisposition.builder("attachment").filename("document.pdf").build()); na headers.add("Content-Disposition", "attachment; filename=\"document.pdf\""); zaczęło pobierać.

Bagietek replied:
Dziwne, u mnie te sposoby działają identycznie.

anastasimars replied:
Może zależy ot przeglądarki

anastasimars replied:
Dodra, u mnie też działa, nie wiem dlaczego ostatnim razem nie zadziałało :)
Można też tak:
headers.setContentDisposition(ContentDisposition.attachment().filename("document.pdf").build());

anastasimars commented on src/main/java/pl/akademiaspecjalistowit/DocumentFlowManagementService/document/exception/DocumentNotFoundException.java:
robię teraz issue z dodaniem exception handlera i u mnie tutaj będzie metoda:
 public DocumentNotFoundException(String message) {
        super(message);
    } i wtedy w serwisie będzie tak: documentDataService.getDocument(documentId).orElseThrow( ()->new DocumentNotFoundException("Document with id " + id + " not found")); jak chcesz, możesz to dodać, lub ja to zrobię w ramach swojego issue.

Bagietek committed:
ASIT-003 correction

Zmiana nazwy getDocumeny na downloadDocument
Dodanie możliwości wpisywania wiadomości do wyjątku DocumentNotFoundException oraz poprawa testu z tym związanego.
